onnx/reference/reference_evaluator.py

```diff
@@ -307,7 +316,31 @@ def _init(self) -> None:
         else:
             self.all_types_ = None  # type: ignore
 
+        original_node_attrs: Dict[str, AttributeProto] = {}
+        if self.original_node is not None:
+            for attr in self.original_node.attribute:
+                original_node_attrs[attr.name] = attr
+            schema = get_schema(
+                self.original_node.op_type,
+                self.opsets_[self.original_node.domain],
+                self.original_node.domain,
+            )
+            for name, attr in schema.attributes.items():
+                original_node_attrs[name] = attr.default_value
+
         for node in self.nodes_:
+            if self.original_node is not None:
+                new_node = NodeProto()
+                new_node.CopyFrom(node)
```

Comment:
I wanted to avoid any copy of protos and have one unique instance of every function. The reason is onnx-script will make an intensive of functions. The model size could be multiplied a lot. I fixed the bug found in #4711 in PR #4710. Class `Constant_12` was supporting attribute from function only for the attribute `value` and not the others. Once fixed, it is working. I originally thought it was because of subgraphs but it was not. PR #4710 fixes it too and other bugs I found while executing every model converted with skl2onnx.

Reply:
Copy cost of patching attr_name seems negligible to me.
Also handling it in operator constructor makes code annoying.
Though it's your design choise so if it's going to be fixed quickly I don't have mind.